index.js

```diff
@@ -6,7 +6,11 @@ module.exports = function (CustomEmitter) {
   var emit = Emitter.prototype.emit
 
   function onevent (packet) {
-    var args = packet.data || []
+    var args = [];
+
+    if (packet.data && packet.data.constructor === Array) {
```

Comment:
Hmmm, random raw data through the websocket should not get through the [parser](https://github.com/socketio/socket.io-protocol#parserdecoder) at all. Maybe what we are dealing with is an [error packet](https://github.com/socketio/socket.io-protocol#error)? In this case, we might actually want to catch that.